src/org/zaproxy/zap/extension/pscan/PassiveScanThread.java

```diff
@@ -141,7 +146,13 @@ public void run() {
 								if (shutDown) {
 									return;
 								}
-								if (scanner.isEnabled() && scanner.appliesToHistoryType(href.getHistoryType())) {
+								int hrefHistoryType = href.getHistoryType();
+								if (pscanOptions.isScanFuzzerMessages()) {
+									addApplicableHistoryType(HistoryReference.TYPE_FUZZER);
+									addApplicableHistoryType(HistoryReference.TYPE_FUZZER_TEMPORARY);
+								}
+								if (scanner.isEnabled() && scanner.appliesToHistoryType(hrefHistoryType)
+										|| getApplicableHistoryTypes().contains(hrefHistoryType)) {
```

Comment:
This should use `optedInHistoryTypes` (to just check additionally the opted-in not all, e.g. the scanner might not want to apply to all of defaults in the first place).